New_PlatformGame/Assets/1. Script/UIFunction.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor.Build.Content;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class UIFunction : MonoBehaviour
+{
+    private int _sceneNumber;
+    private void Start()
+    {
+        _sceneNumber = SceneManager.GetActiveScene().buildIndex;
+    }
+
+    public void GoBackScene()
+    {
+        SceneManager.LoadScene(_sceneNumber - 1);
+    }
+
+    public void ReloadScene()
+    {
+        SceneManager.LoadScene(_sceneNumber);
+    }
+
+    public void GoNextScene()
+    {
+        SceneManager.LoadScene(_sceneNumber + 1);
+    }
+
+    public void EndGame()
+    {
+#if UNITY_EDITOR
+        UnityEditor.EditorApplication.isPlaying = false;
+#else
+         Application.Quit();
+#endif
```

Comment:
```suggestion
#if UNITY_EDITOR
        UnityEditor.EditorApplication.isPlaying = false;
#else
         Application.Quit();
#endif
```
이런식으로 전처리 구문 넣어주는 습관 좋음. 컴파일 후 빌드 파일이 덜 무거워지도록 하게 도와주는 센스 좋음.
Debug.Log같은것도 에디터에서만 돌아가니 넣어주면 빌드 파일 퍼포먼스에 도움이 될 거고,
https://docs.unity3d.com/kr/530/Manual/PlatformDependentCompilation.html
본 링크를 참고해서 유니티에서 지원하는 전처리문을 확인할 수 있음.